cloud/docker/docker_login.py

```diff
@@ -0,0 +1,246 @@
+#!/usr/bin/python
+#
+
+# (c) 2015, Olaf Kilian <olaf.kilian@symanex.com>
+#
+# This file is part of Ansible
+#
+# This module is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This software is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this software.  If not, see <http://www.gnu.org/licenses/>.
+
+######################################################################
+
+DOCUMENTATION = '''
+---
+module: docker_login
+author: Olaf Kilian
+version_added: "2.0"
+short_description: Manage Docker registry logins
+description:
+     - Ansible version of the "docker login" CLI command.
+     - This module allows you to login to a Docker registry without directly pulling an image or performing any other actions.
+     - It will write your login credentials to your local .dockercfg file that is compatible to the Docker CLI client as well as docker-py and all other Docker related modules that are based on docker-py.
+options:
+  registry:
+    description:
+       - URL of the registry, for example: https://index.docker.io/v1/
+    required: true
+  username:
+    description:
+       - The username for the registry account
+    required: true
+  password:
+    description:
+       - The plaintext password for the registry account
+    required: true
+  email:
+    description:
+       - The email address for the registry account
+    required: false
+    default: anonymous@localhost.local
+  reauth:
+    description:
+       - Whether refresh existing authentication on the Docker server (boolean)
+    required: false
+    default: false
+  dockercfg_path:
+    description:
+       - Use a custom path for the .dockercfg file
+    required: false
+    default: ~/.dockercfg
+  docker_url:
+    descriptions:
+       - Refers to the protocol+hostname+port where the Docker server is hosted
+    required: false
+    default: unix://var/run/docker.sock
+  timeout:
+    description:
+       - The HTTP request timeout in seconds
+    required: false
+    default: 600
+
+requirements: [ "docker-py" ]
```

Comment:
Add a requirements "python >= 2.6"

Since docker-py requires newer python (that lets us know that we don't have to make this module compatible with pyhton-2.4.)